src/Sql/Sql/ManagedInstance/Model/AzureSqlManagedInstanceModel.cs

```diff
@@ -174,5 +174,10 @@ public class AzureSqlManagedInstanceModel
         /// Gets or sets whether or not the multi-az is enabled
         /// </summary>
         public bool? ZoneRedundant { get; set; }
+
+        /// <summary>
+        /// Gets or sets the service principal object
+        /// </summary>
+        public Management.Sql.Models.ServicePrincipal ServicePrincipal { get; set; }
```

Comment:
AzureSqlManagedInstanceModel is cmdlet output. This type is from SDK directly, please wrap it to avoid breaking change in the future.